docs/data/data-grid/row-grouping/row-grouping.md

```diff
@@ -142,9 +142,9 @@ return (
 ```
 
 :::warning
-This hook is only compatible with the deprecated column property `hide` or with the controlled `columnVisibilityModel` prop.
+This hook is not compatible with the deprecated column property `hide`, and you should not control the `columnVisibilityModel`.
```

Comment:
Wow my sentence made no sense...
It should be compatible with the controlled `columnVisibilityModel` prop if you also pass its value to the hook.

See the `RowGroupingCustomGroupingColDefCallback` example